tests/test_table.py

```diff
@@ -126,22 +126,37 @@ def test_query(TestModel, TestModel_entries, dynamo_local):
 
 def test_scan(TestModel, TestModel_entries, dynamo_local):
     """Scanning should return the expected values"""
-    results = TestModel.scan(count__gt=200)
+    results = list(TestModel.scan(count__gt=200))
     assert len(results) == 2
 
     # our table has a hash and range key, so our results are ordered based on the range key
     assert results[0].count == 333
     assert results[1].count == 222
 
-    results = TestModel.scan(child__sub="two")
+    results = list(TestModel.scan(child__sub="two"))
     assert len(results) == 1
     assert results[0].count == 222
 
-    results = TestModel.scan(child__sub__begins_with="t")
+    results = list(TestModel.scan(child__sub__begins_with="t"))
     assert len(results) == 2
     assert results[0].count == 333
     assert results[1].count == 222
 
+    # Test auto-paging
+    results = list(TestModel.scan(scan_kwargs={'Limit': 2}, count__gt=0))
+    assert len(results) == 3
```

Comment:
This is an interesting use case...

If I set a limit of 2 and got 3 results back I would be a little surprised.  Based on [the Scan Limit docs](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.Limit) I feel like we should disable auto-paging if a limit was supplied.

Given:

> Now suppose you add a filter expression to the Scan. In this case, DynamoDB will apply the filter expression to the six items that were returned, discarding those that do not match. The final Scan result will contain 6 items or fewer, depending on the number of items that were filtered.

I feel like we should always return `<=N`, where `N` is the limit.

What do you think?